website/versioned_docs/version-0.10/database-schema-migrations.md

```diff
@@ -0,0 +1,208 @@
+---
+id: version-0.10-database-schema-migrations
+title: Database Schema Migrations
+original_id: database-schema-migrations
+---
+
+## Database Schema Migrations
+
+Graphback uses [graphql-migrations](../packages/graphql-migrations) to automatically create and update tables from a GraphQL schema.
+
+### CLI
+
+To create or update your database from the CLI, run:
+
+```sh
+graphback db
+```
+
+### Usage
+
+The `migrateDB` method creates and updates your tables and columns to match your GraphQL schema.
+
+All the database operations are wrapped in a single transaction, so your database will be fully rolled back to its initial state if an error occurs.
+
+```ts
+import * as jsonConfig from '../graphback.json'
+import { schemaText } from './schema';
+import { migrateDB } from 'graphql-migrations';
+import { GraphQLBackendCreator, PgKnexDBDataProvider } from 'graphback';
+
+const backend = new GraphQLBackendCreator(schemaText, jsonConfig.graphqlCRUD);
+const dbClientProvider = new PgKnexDBDataProvider(client);
+
+const dbConfig = {
+  client: jsonConfig.db.database,
+  connection: jsonConfig.db.dbConfig
+};
+
+migrateDB(dbConfig, schemaText, {
+  // Additional options
+}).then((ops) => {
+    console.log(ops);
+});
+
+const pubSub = new PubSub();
+
+const runtime = await backend.createRuntime(dbClientProvider, pubSub);
+...
+```
+
+### Migration Options
+
+- `config`: database configuration options.
+- `schemaText`: GraphQL schema text.
+- `options`:
+  - `updateComments`: overwrite comments on table and columns.
+  - `scalarMap`: Custom scalar mapping. Default: `null`.
+  - `mapListToJson`: Map scalar lists to JSON column type by default.
+  - `debug`: display debugging information and SQL queries.
+  - `removeDirectivesFromSchema`: removes all directives from the GraphQL schema.
+  - All other options are not currently supported by Graphback.
+
+### Defining your data model
+
+#### Primary key
+
+Each type must have a primary key. The primary key field must be `id` and the type must be `ID`.
+
+```gql
+type Note {
+  id: ID!
+  title: String!
+}
+```
+
+#### Foreign key
+
+To set a foreign key, set a field reference to the related type.
+
+```gql
+type Comment {
+  id: ID!
+  note: Note! # this creates a `noteId` column in the `comment` table.
+}
+
+type Note {
+  id: ID!
+  title: String!
+}
+```
+
+> NOTE: The field name must match that of the related type. Support for custom field names is coming soon.
```

Comment:
This is out of date. This one is on me.

I've provided a PR o update this here https://github.com/aerogear/graphback/pull/609